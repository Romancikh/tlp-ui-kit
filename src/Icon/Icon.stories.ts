import type { Meta, StoryObj } from "@storybook/react";

import Icon from "./Icon";

const meta = {
  title: "Icon",
  component: Icon,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    color: { control: "select" },
  },
} satisfies Meta<typeof Icon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const AddIcon: Story = {
  args: {
    name: "AddIcon",
  },
};

export const Colored: Story = {
  args: {
    name: "AddIcon",
    color: "success",
  },
};

export const AutoStoriesIcon: Story = {
  args: {
    name: "AutoStoriesIcon",
  },
};

export const CancelOutlinedIcon: Story = {
  args: {
    name: "CancelOutlinedIcon",
  },
};

export const CheckCircleOutlineIcon: Story = {
  args: {
    name: "CheckCircleOutlineIcon",
  },
};

export const DeleteOutlineIcon: Story = {
  args: {
    name: "DeleteOutlineIcon",
  },
};

export const EditIcon: Story = {
  args: {
    name: "EditIcon",
  },
};

export const MenuIcon: Story = {
  args: {
    name: "MenuIcon",
  },
};

export const NavigateBeforeIcon: Story = {
  args: {
    name: "NavigateBeforeIcon",
  },
};
