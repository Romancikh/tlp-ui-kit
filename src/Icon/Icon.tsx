import AddIcon from "@mui/icons-material/Add";
import AutoStoriesIcon from "@mui/icons-material/AutoStories";
import CancelOutlinedIcon from "@mui/icons-material/CancelOutlined";
import CheckCircleOutlineIcon from "@mui/icons-material/CheckCircleOutline";
import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
import EditIcon from "@mui/icons-material/Edit";
import MenuIcon from "@mui/icons-material/Menu";
import NavigateBeforeIcon from "@mui/icons-material/NavigateBefore";

const icons = {
  AddIcon,
  AutoStoriesIcon,
  CancelOutlinedIcon,
  CheckCircleOutlineIcon,
  DeleteOutlineIcon,
  EditIcon,
  MenuIcon,
  NavigateBeforeIcon,
};

type IconProps = {
  name: keyof typeof icons;
  color?: "success" | "primary" | "error" | "secondary";
};

function Icon({ name, color }: IconProps) {
  const SelectedIcon = icons[name];
  return <SelectedIcon color={color} />;
}

export default Icon;
