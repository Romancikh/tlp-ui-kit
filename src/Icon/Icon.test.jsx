import { render } from "@testing-library/react";
import Icon from "./Icon";

test("AddIcon renders correctly", () => {
  const { container } = render(<Icon name="AddIcon" />);
  expect(container).toMatchSnapshot();
});

test("Colored AddIcon renders correctly", () => {
  const { container } = render(<Icon name="AddIcon" color="success" />);
  expect(container).toMatchSnapshot();
});

test("AutoStoriesIcon renders correctly", () => {
  const { container } = render(<Icon name="AutoStoriesIcon" />);
  expect(container).toMatchSnapshot();
});

test("CancelOutlinedIcon renders correctly", () => {
  const { container } = render(<Icon name="CancelOutlinedIcon" />);
  expect(container).toMatchSnapshot();
});

test("CheckCircleOutlineIcon renders correctly", () => {
  const { container } = render(<Icon name="CheckCircleOutlineIcon" />);
  expect(container).toMatchSnapshot();
});

test("DeleteOutlineIcon renders correctly", () => {
  const { container } = render(<Icon name="DeleteOutlineIcon" />);
  expect(container).toMatchSnapshot();
});

test("EditIcon renders correctly", () => {
  const { container } = render(<Icon name="EditIcon" />);
  expect(container).toMatchSnapshot();
});

test("MenuIcon renders correctly", () => {
  const { container } = render(<Icon name="MenuIcon" />);
  expect(container).toMatchSnapshot();
});

test("NavigateBeforeIcon renders correctly", () => {
  const { container } = render(<Icon name="NavigateBeforeIcon" />);
  expect(container).toMatchSnapshot();
});
