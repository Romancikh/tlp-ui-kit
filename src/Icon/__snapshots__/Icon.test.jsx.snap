// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="AddIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</div>
`;

exports[`AutoStoriesIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="AutoStoriesIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
    />
  </svg>
</div>
`;

exports[`CancelOutlinedIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="CancelOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
    />
  </svg>
</div>
`;

exports[`CheckCircleOutlineIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="CheckCircleOutlineIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
</div>
`;

exports[`Colored AddIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
    data-testid="AddIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</div>
`;

exports[`DeleteOutlineIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="DeleteOutlineIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
    />
  </svg>
</div>
`;

exports[`EditIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="EditIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
    />
  </svg>
</div>
`;

exports[`MenuIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="MenuIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
</div>
`;

exports[`NavigateBeforeIcon renders correctly 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="NavigateBeforeIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
    />
  </svg>
</div>
`;
